src/string/CreditCard.ts

```diff
@@ -0,0 +1,193 @@
+/**
+ * Represents (some) valid credit card numbers.
+ *
+ * At the moment, this mostly handles Visa, Mastercard, American Express, Diners Club,
+ * Discover, and JCB.
+ *
+ * @since 0.0.3
+ */
+import { Kind, Kind2, URIS, URIS2, HKT2 } from 'fp-ts/HKT'
+import * as D from 'io-ts/Decoder'
+import * as Enc from 'io-ts/Encoder'
+import * as Eq_ from 'fp-ts/Eq'
+import * as G from 'io-ts/Guard'
+import * as Str from 'fp-ts/string'
+import * as TD from 'io-ts/TaskDecoder'
+import * as t from 'io-ts/Type'
+import { pipe } from 'fp-ts/function'
+import * as fc from 'fast-check'
+
+import * as Arb from '../internal/ArbitraryBase'
+import { digits } from '../internal/util'
+import { luhn } from '../internal/algorithms'
+
+/**
+ * @since 0.0.3
+ * @category Internal
+ */
+interface CreditCardBrand {
+  readonly CreditCard: unique symbol
+}
+
+/**
+ * Represents (some) valid credit card numbers.
+ *
+ * At the moment, this mostly handles Visa, Mastercard, American Express, Diners Club,
+ * Discover, and JCB.
+ *
+ * @since 0.0.3
+ * @category Model
+ */
+export type CreditCard = string & CreditCardBrand
+
+/**
+ * @since 0.0.3
+ * @category Model
+ */
+export type SchemableParams<S> = HKT2<S, string, CreditCard>
+
+/**
+ * @since 0.0.3
+ * @category Model
+ */
+export type SchemableParams1<S extends URIS> = Kind<S, CreditCard>
+
+/**
+ * @since 0.0.3
+ * @category Model
+ */
+export type SchemableParams2<S extends URIS2> = Kind2<S, string, CreditCard>
+
+/**
+ * @since 0.0.3
+ * @category Model
+ */
+export type SchemableParams2C<S extends URIS2> = Kind2<S, unknown, CreditCard>
+
+const cc = new RegExp(
```

Comment:
:exclamation: this is now totally custom & cited

(ps checking for _ranges_ of numbers in a regex is a PITA)

Reply:
💯 I had a feeling CC would be a huge pain! 😁 nonetheless, I appreciate the thoroughness!